gtsam/CMakeLists.txt

```diff
@@ -101,6 +101,15 @@ message(STATUS "Building GTSAM - shared: ${BUILD_SHARED_LIBS}")
 
 # BUILD_SHARED_LIBS automatically defines static/shared libs:
 add_library(gtsam ${gtsam_srcs})
+
+if(NOT CMAKE_VERSION VERSION_LESS 3.8)
+	target_compile_features(gtsam PUBLIC cxx_std_11)
+else()
+	if (NOT MSVC)
+		target_compile_options(gtsam PUBLIC -std=c++11)
```

Comment:
Unfortunately this also doesn't seem to work. Failed in Travis, and for me, too. I have CMake 3.5.1, but `-std=c++11` does not appear in my compiler args. I checked that we *do* reach this line, so there's some problem with actually setting the property on the target?  First error:
```
[  2%] Building CXX object wrap/CMakeFiles/wrap_lib.dir/Method.cpp.o
cd /home/cbeall/git/gtsam/build-scratch/wrap && ccache /usr/bin/c++    -I/home/cbeall/git/gtsam  -Wall -O3 -DNDEBUG    -O3 -DNDEBUG   -o CMakeFiles/wrap_lib.dir/Method.cpp.o -c /home/cbeall/git/gtsam/wrap/Method.cpp
In file included from /usr/include/c++/5/unordered_set:35:0,
                 from /home/cbeall/git/gtsam/wrap/OverloadedFunction.h:23,
                 from /home/cbeall/git/gtsam/wrap/FullyOverloadedFunction.h:21,
                 from /home/cbeall/git/gtsam/wrap/MethodBase.h:22,
                 from /home/cbeall/git/gtsam/wrap/Method.h:21,
                 from /home/cbeall/git/gtsam/wrap/Method.cpp:18:
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^
```
It also doesn't show up when we print config options (`march=native` is also missing):
```
--   C compilation flags            :  -Wall -O3 -DNDEBUG    -O3 -DNDEBUG
--   C++ compilation flags          :  -Wall -O3 -DNDEBUG    -O3 -DNDEBUG
```

Reply:
Ok, I had to log in into a u16.04 docker image to debug this one, but I think it should be fixed now ;-)

All commits in this PR have been rebased for ease of history inspection.

PS: CXX_FLAGS won't display all the flags now, since most flags are now set via cmake target properties. So, those properties are now also shown in the console in the final cmake summary. There are "scary" cmake expressions over there (`$<condition:...>`), but that's right, and it's one of the main reasons of why using this system (+exporting cmake targets) makes a system easier to maintain in the long term (hope so!).